extra/modules/confiant-ad-quality/src/main/java/org/prebid/server/hooks/modules/com/confiant/adquality/v1/ConfiantAdQualityBidResponsesScanHook.java

```diff
@@ -90,8 +106,12 @@ private InvocationResult<AllProcessedBidResponsesPayload> toInvocationResult(
                         .debugMessages(debugEnabled
                                 ? bidsScanResult.getDebugMessages()
                                 : null)
+                        .analyticsTags(AnalyticsMapper.toAnalyticsTags(
+                                bidsScanResult, scannedBidderResponses, notScannedBidderResponses))
                         .payloadUpdate(payload -> hasIssues
-                                ? AllProcessedBidResponsesPayloadImpl.of(bidsScanResult.filterValidResponses(payload.bidResponses()))
+                                ? AllProcessedBidResponsesPayloadImpl.of(
+                                    Stream.concat(bidsScanResult.filterValidResponses(scannedBidderResponses).stream(),
+                                            notScannedBidderResponses.stream()).toList())
```

Comment:
`scannedBidderRepsponses` are filtered twice implicitly (in the `toAnalyticsTags` and here), I believe `bidsScanResult.filterValidResponses(scannedBidderResponses)` can be done once and the result can be passed to the `toAnalyticsTags` (and I believe it should be filtered only if the `bidsScanResult.hasIssues()` returns true)